### PR TITLE
Add drag-and-drop reordering for admin menu items

### LINKED ISSUE #9: Add drag-and-drop reordering to menu items
Currently, the menu items can only be disabled in the user-admin-simplifier plugin. To enhance usability, add the ability to reorder menu items via drag-and-drop in the settings UI, in addition to the current option to disable them.

**Requested enhancement:**
- Implement drag-and-drop reordering for menu items in the user-admin-simplifier interface. 
- Ensure the new order is correctly persisted and reflected in the user menu
- Maintain the ability to disable items alongside reordering.

**Why it's needed:**
This will make the menu customizations much more flexible and improve user experience for site administrators.

### PR DESCRIPTION
**Pull request by adamsilverstein**
Fixes #9 (https://github.com/adamsilverstein/user-admin-simplifier/issues/9)

Adds drag-and-drop reordering of top-level admin menu items to the settings UI, alongside the existing enable/disable checkboxes, and applies the saved order to the targeted user's actual WordPress admin menu.

## UI

- Each top-level menu row in the settings screen now has a drag handle (native HTML5 drag-and-drop, no new runtime dependencies) plus keyboard accessible move up/down buttons with per-item aria labels, so reordering is not mouse-only.
- Dragging is initiated from the handle only, so checkboxes, labels and the submenu toggle keep working as before.
- The list renders in the selected user's saved order; a short hint explains the feature.

## Storage

- The saved order is stored as a `menu-order` sub-key in the existing per-user options array: an ordered list of `sanitize_key()`'d menu ids.
- The AJAX save handler pulls `menu-order` out of the payload before the existing `intval()` flag sanitization, and sanitizes it via a new `uas_sanitize_menu_order()` helper (type filtering, `sanitize_key()`, deduping).
- Fully backward compatible: users with no saved order keep the default menu order, existing disable flags are untouched, and "Reset User Settings" clears the order too.

## Menu application

- A new pure helper, `uas_apply_menu_order()`, reorders the global `$menu` array in `uas_edit_admin_menus()` (admin_head), after disabled menus are removed.
- The reorder is a stable slot refill: items present in the saved order are rearranged to match it, while separators and items not in the saved order (e.g. newly installed plugins) keep their default relative positions. Original menu position keys are preserved.

## Tests

- New `tests/test-menu-order.php` (13 cases) covering order sanitization and reordering edge cases (separators stay in place, unknown ids skipped, duplicate/invalid keys dropped, position keys preserved), following the existing standalone CLI test style.
- Wired into `npm run test:php` and the CI PHP test matrix.
- Verified locally: both PHP test suites pass, `php -l` clean, PHPStan (level 5) clean, ESLint clean, production webpack build succeeds (rebuilt `build/` artifacts committed, as on main).

No version bump, per the in-flight 3.0.1 release PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Menu reordering: Users can now drag-and-drop admin menu items to customize their order.
  * Keyboard accessibility: Up/down buttons enable menu reordering without a mouse.
  * Per-user persistence: Custom menu ordering is automatically saved for each user.
  * Enhanced drag-and-drop UI with visual feedback including drag handles, move buttons, and drop indicators.

* **Tests**
  * Added comprehensive test coverage for menu ordering functionality.

### CODE REVIEW

**Comment by adamsilverstein** [comment]
@coderabbitai review

File: tests/test-menu-order.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+			continue;
+		}
+		$menu_id = sanitize_key( $menu_id );
+		if ( '' !== $menu_id && ! in_array( $menu_id, $sanitized, true ) ) {
+			$sanitized[] = $menu_id;
+		}
+	}
+
+	return $sanitized;
+}
```
[the remark this answers is not in the document]
Comment (adamsilverstein) [reply tests/test-menu-order.php:58]: The duplication is deliberate and follows the existing convention in this suite - tests/test-menu-name-cleaning.php copies uas_clean_menu_name() the same way, because requiring useradminsimplifier.php would need stubs for all the top-level WordPress calls (add_action, get_option, etc.). Consolidating the test bootstrap so tests load the production file is a worthwhile follow-up, but out of scope for this feature PR.
